include/alpaka/math/abs/AbsUniformCudaHipBuiltIn.hpp

```diff
@@ -83,6 +84,16 @@ namespace alpaka
                     return ::fabsf(arg);
                 }
             };
+            //! The CUDA built in abs Complex specialization.
+            template<typename T>
+            struct Abs<AbsUniformCudaHipBuiltIn, Complex<T>>
+            {
+                __device__ auto operator()(AbsUniformCudaHipBuiltIn const& abs_ctx, Complex<T> const& arg)
+                {
+                    alpaka::ignore_unused(abs_ctx);
+                    return ::sqrt(arg.real() * arg.real() + arg.imag() * arg.imag());
+                }
+            };
```

Comment:
Since core CUDA doesn't have any complex type (only CUDA libraries do), we will have to just implement everything manually

Reply:
Probably `::sqrt` is wrong for `T == float`, and I should have called alpaka's `sqrt()` instead

Reply:
Do you mean with cuda complex type a class or struct (I think so)? There are a lot of complex types in CUDA, but every complex type is a alias of float2.

Reply:
There are in CUDA libraries, not in CUDA itself as far as I understand

Reply:
Attention, there are more complex types in the CUDA libraries, e.g. `cuFFT`
```C
typedef cuComplex cufftComplex;
typedef cuDoubleComplex cufftDoubleComplex;
```

But should be all a typedef of `float2`.